build.sh

```diff
@@ -1,77 +1,27 @@
 #!/usr/bin/env bash
```

Comment:
I get an error running this script now; previous behavior is to default to building when invoked with no arguments.

```bash
-> DocoptException: errors Error: Error in Argument 1 ('<>')
   Expecting: Command 'build', Command 'run', Flag '--help', Flag '--version' or
   ShortFlag '-h'
```

Reply:
Further, when passing the `build` argument to this script:

```
---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target         Duration
------         --------
AssemblyInfo   00:00:01.9929564
GenerateHelp   00:00:01.9941933   (generating reference documentation failed: [])
Build          00:00:00           (skipped)
CopyBinaries   00:00:00           (skipped)
RunTests       00:00:00           (skipped)
GenerateDocs   00:00:00           (skipped)
NuGet          00:00:00           (skipped)
BuildPackage   00:00:00           (skipped)
All            00:00:00           (skipped)
Total:         00:00:04.1852372
Status:        Failure
---------------------------------------------------------------------
Script reported an error:
-> BuildFailedException: Target 'GenerateHelp' failed.
-> One or more errors occurred. (generating reference documentation failed: [])
-> generating reference documentation failed: []
```

Reply:
For the first issue: the command line interface of fake has changed starting from v5, please visit: https://fake.build/fake-commandline.html#Basic-examples
basically you should invoke `fake run build.fsx --target x` or `fake build --target x`. If you omit the target, it will run the default target.
As for generate.fsx issue, you have to run paket install before running the build. However, i would recommend getting rid of Fsharp.formatting based documentation completely, because:
- that's the only item that ties us to the existence of a mono
- fragile, cannot be migrated as 1-1
- the generated documentation is not used at all. I mean, it has no much content in it and it is not published anywhere.

Please also note that both fake and paket are now dotnet tools which can be installed with:
```
dotnet tool install --global fake-cli
dotnet tool install --global paket
```

Reply:
If that's the case, what is the purpose of `build.sh` in this branch?

Reply:
We're not using said documentation, I'd be in favor of removing it from the build.

Reply:
In other words: `build.sh` is currently the entrypoint that performs the whole build, by default, with no arguments.  I either need this functionality restored, or a concise recipe for what replaces it.

Reply:
> If that's the case, what is the purpose of `build.sh` in this branch?

Basically, it hides the fact that fake is a dotnet cli command from you. It downloads fake for you and passes your provided arguments to it.
The FAKE template comes with it so i thought i retain it for the future as it serves some similar purpose as the old build.sh.
It is here: https://github.com/fsharp/FAKE/blob/e57683b2dfe0fa856502326f308a3a23670ddf46/src/template/fake-template/Content/fake.tool.sh


> We're not using said documentation, I'd be in favor of removing it from the build.

Happy to remove it. See the upcoming commit. Are you okay if i do the same in the Amyris.Bio migration as well?


> In other words: `build.sh` is currently the entrypoint that performs the whole build, by default, with no arguments. I either need this functionality restored, or a concise recipe for what replaces it.

As per my understanding, if you invoke `./build.sh build` or `fake build` without any arguments, it will still run your default target. Is that good for you?

Reply:
> > We're not using said documentation, I'd be in favor of removing it from the build.
> 
> Happy to remove it. See the upcoming commit. Are you okay if i do the same in the Amyris.Bio migration as well?

Done. please see my very last commit.

Reply:
Onwards to

```
~/src/GslCore$ ./build.sh build
Paket version 5.206.0
The last restore is still up to date. Nothing left to do.
Performance:
 - Runtime: 196 milliseconds
Package Fake.Core.Target is defined more than once in group Main of /home/macklin/src/GslCore/.fake/build.fsx/paket.dependencies
Performance:
 - Cli parsing: 462 milliseconds
 - Packages: 361 milliseconds
 - Script analyzing: 367 milliseconds
 - Runtime: 1 second
There was a problem while setting up the environment:
-> Error during parsing of '/home/macklin/src/GslCore/.fake/build.fsx/paket.lock'.
-> Can't parse ">=". System.ArgumentException: SemVer.Parse
   Parameter name: version
      at Paket.SemVer.Parse$cont@269-1(FSharpList`1 fragments, Unit unitVar)
      at Paket.SemVer.Parse$cont@269(FSharpList`1 fragments, Unit unitVar)
      at Paket.SemVer.f@62-1(String version)
```

Reply:
what is your .fake/fake --version ?

Reply:
```
~/src/GslCore$ .fake/fake --version
FAKE 5 - F# Make (5.13.5) (this line is written to standard error, see https://github.com/fsharp/FAKE/issues/2066)
FakePath: /home/macklin/src/GslCore/.fake/.store/fake-cli/5.13.5/fake-cli/5.13.5/tools/netcoreapp2.1/any/Fake.Runtime.dll
Paket.Core: 5.203.2
```

Reply:
May i ask you to do these steps and try again:
* remove src/GslCore/bin
* remove src/GslCore/obj
* remove tests/GslCore.Tests/bin
* remove tests/GslCore.Tests/obj
*  .paket/paket clear-cache --clear-local
* remove paket-files
* remove .paket
* remove .fake
* remove bin

I know this is basically burning all bridges, but sometimes these tools tend to have something stuck in the pipeline. I am sure it should be something like that because i am trying to do the same in a fresh linux installation with the same tool versions and it works without issues

Reply:
I started from a totally clean environment and still ran into the same SemVer issue above.

Reply:
(This is on Ubuntu 16.04)

Reply:
Unfortunately, i cannot reproduce your issue.
I have pinned the tool + tool package versions in a new commit. Could you repeat the steps above after a pull, please?

Now i tested the branch on Windows 10, Linux Mint 19 and on a vanilla Ubuntu 16.04 using docker.
This builds without any issues:
```docker
FROM ubuntu:16.04

RUN apt-get update && apt-get install -y wget apt-transport-https
RUN wget -q https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb
RUN dpkg -i packages-microsoft-prod.deb
RUN apt-get update
RUN apt-get install -y apt-transport-https
RUN apt-get install -y dotnet-sdk-2.2 git
RUN git clone https://github.com/legezam/GslCore.git
WORKDIR GslCore
RUN git fetch
RUN git checkout dotnetcore
RUN ./build.sh build
CMD ["bash"]
```

Reply:
Thanks for trying to reproduce, I eventually tracked it down to a network-mounted filesystem issue that I'm working to fix now.

Reply:
Great! Besides being a repro, this image could serve as a gitlab ci image
for this repo :)